spock-core/src/main/java/org/spockframework/runtime/extension/builtin/TempDirInterceptor.java

```diff
@@ -0,0 +1,112 @@
+package org.spockframework.runtime.extension.builtin;
+
+import org.codehaus.groovy.runtime.ResourceGroovyMethods;
+import org.spockframework.runtime.extension.IMethodInterceptor;
+import org.spockframework.runtime.extension.IMethodInvocation;
+import org.spockframework.runtime.model.FieldInfo;
+import org.spockframework.runtime.model.MethodKind;
+import org.spockframework.util.Beta;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+/**
+ * @author dqyuan
+ * @since 2.0
+ */
+@Beta
+public class TempDirInterceptor implements IMethodInterceptor {
+  private static final String TEMP_DIR_PREFIX = "spock";
+
+  private final Class<?> fieldType;
+  private final FieldInfo fieldInfo;
+  private final Path parentDir;
+  private final MethodKind interceptPoint;
+  private final boolean keep;
+
+  private Path tempPath;
+
+  TempDirInterceptor(Class<?> fieldType, FieldInfo fieldInfo,
+                     MethodKind interceptPoint, Path parentDir,
+                     boolean keep) {
+    this.fieldType = fieldType;
+    this.fieldInfo = fieldInfo;
+    this.parentDir = parentDir;
+    this.interceptPoint = interceptPoint;
+    this.keep = keep;
+  }
+
+  private Path generateTempDir(IMethodInvocation invocation) throws IOException {
+    String prefix = TEMP_DIR_PREFIX + "_" + ((invocation.getIteration() == null) ?
+      invocation.getSpec().getName() : invocation.getIteration().getName())
+      .replaceAll("[^a-zA-Z0-9_.-]++", "_") + "_" + fieldInfo.getName();
```

Comment:
Nice regexp :-).
(assuming it works as expected ;-) ).

Reply:
It just replaces everything that is not alnum, underscore, dot and dash with underscore, so it should be fine if I didn't mistype :-D